src/Plugin.php

```diff
@@ -24,7 +24,7 @@
      * @param callable         $next    Next middleware in the chain, the request is passed as the first argument
      * @param callable         $first   First middleware in the chain, used to to restart a request
      *
-     * @return Promise
+     * @return Promise Resolves a PSR-7 Response or fails with an Http\Client\Exception (The same as HttpAsyncClient).
```

Comment:
We don't but we should. Write 
`@see http://php-http.readthedocs.io/en/latest/plugins/introduction.html`

FYI: https://github.com/php-http/documentation/issues/149